pulp_smash/tests/pulp3/file/api_v3/test_crud_content_unit.py

```diff
@@ -0,0 +1,94 @@
+
+    This test targets the following issues:
+
+    * `Pulp #2872 <https://pulp.plan.io/issues/2872>`_
+    * `Pulp Smash #870 <https://github.com/PulpQE/pulp-smash/issues/870>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variable."""
+        cls.artifact = {}
+        cls.cfg = config.get_config()
+        clean_artifacts(cls.cfg)
+        cls.content_unit = {}
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        files = {'file': utils.http_get(FILE_URL)}
+        cls.artifact = cls.client.post(ARTIFACTS_PATH, files=files)
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean class-wide variable."""
+        cls.client.delete(cls.artifact['_href'])
+
+    def test_01_create_content_units(self):
+        """Create content unit."""
+        attrs = _gen_verbose_content_unit_attrs(self.artifact, utils.uuid4())
+        type(self).content_unit = self.client.post(
+            FILE_CONTENT_PATH,
+            data=attrs
+        )
+
+    @selectors.skip_if(bool, 'content_unit', False)
+    def test_02_read_content_unit(self):
+        """Read a content unit by its href."""
+        content_unit = self.client.get(self.content_unit['_href'])
+        for key, val in self.content_unit.items():
+            with self.subTest(key=key):
+                self.assertEqual(content_unit[key], val)
+
+    @selectors.skip_if(bool, 'content_unit', False)
+    def test_03_partially_update(self):
+        """Attempt to update a content unit using HTTP PATCH.
+
+        This HTTP method is not supported and a HTTP exception is expected.
+        """
+        attrs = _gen_verbose_content_unit_attrs(self.artifact, utils.uuid4())
+        with self.assertRaises(HTTPError):
+            self.client.patch(self.content_unit['_href'], data=attrs)
+
+    @selectors.skip_if(bool, 'content_unit', False)
+    def test_03_fully_update(self):
+        """Attempt to update a content unit using HTTP PUT.
+
+        This HTTP method is not supported and a HTTP exception is expected.
+        """
+        attrs = _gen_verbose_content_unit_attrs(self.artifact, utils.uuid4())
+        with self.assertRaises(HTTPError):
+            self.client.put(self.content_unit['_href'], data=attrs)
+
+    @selectors.skip_if(bool, 'content_unit', False)
+    def test_04_delete_content_unit(self):
+        """Delete a content unit."""
+        self.client.delete(self.content_unit['_href'])
+        with self.assertRaises(HTTPError):
+            self.client.get(self.content_unit['_href'])
+
+
+def _gen_verbose_content_unit_attrs(artifact, path):
+    """Generate a dict with content unit attributes.
+
+    :param: artifact: A dict about the artifact.
+    :param: path: The path in which one the content unit will be available.
+    """
+    return {
+        'artifacts': {path: artifact['file']},
+        'digest': artifact['sha256'],
+        'path': path
```

Comment:
Just a heads up: these fields are changing. It's going to be just `artifact` and `relative_path`. See https://github.com/pulp/pulp_file/pull/51.